lib/translator/inquiry.cc

```diff
@@ -13,12 +13,75 @@
 // limitations under the License.
 
 #include "inquiry.h"
+
 #include "string.h"
 
 namespace translator {
 
-// command specific helpers
+// anonymous namespace for helper functions and constant expressions
 namespace {
+
+// Builds the IdentificationDescriptor according to 6.1.4.5 of the translation
+// spec:
+// https://nvmexpress.org/wp-content/uploads/NVM_Express_-_SCSI_Translation_Reference-1_5_20150624_Gold.pdf
+// This only builds the constant part of the Identification Descriptor. The
+// caller is responsible for building the identifier.
+scsi_defs::IdentificationDescriptor BuildIdentificationDescriptor(
+    const nvme_defs::IdentifyNamespace& identify_namespace_data) {
+  // check if nonzero
+  bool nguid_nz =
+      identify_namespace_data.nguid[0] || identify_namespace_data.nguid[1];
+  bool eui64_nz = identify_namespace_data.eui64;
+  uint8_t identifier_length;
+
+  if (nguid_nz) {
+    // Shall be set to 08h if the EUI-64 field is populated from the NVMe EUI64
+    // field.
+    identifier_length = kIdentifierLengthNguid;
+
+    uint8_t identification_descriptor_identifier[16];
+
+    memcpy(&identification_descriptor_identifier,
```

Comment:
Why do we memcpy this data into identification_descriptor_identifier and do nothing with it

Reply:
This function returns the identificaiton descriptor to the `TranslateDeviceIdentification` [function](https://github.com/googleinterns/scsi2nvme/pull/128/commits/5fc032b0e0019378dcf8d24be631e780c13b5480) which then [writes](https://github.com/googleinterns/scsi2nvme/pull/128/commits/5fc032b0e0019378dcf8d24be631e780c13b5480) it to the `buffer`.

Reply:
Yup, I was doing something that didn't work. I've updated the code now. Let me know your thoughts